mp/src/game/server/momentum/mom_system_saveloc.cpp

```diff
@@ -260,11 +258,43 @@ void CSaveLocSystem::LevelInitPreEntity()
     }
 }
 
+bool CSaveLocSystem::LoadStartMarks()
+{
+    if (!m_pSavedLocsKV || m_pSavedLocsKV->IsEmpty())
+        return false;
+
+    KeyValues *kvMapSavelocs = m_pSavedLocsKV->FindKey(gpGlobals->mapname.ToCStr());
+    if (!kvMapSavelocs || kvMapSavelocs->IsEmpty())
+        return false;
+
+    KeyValues *kvStartMarks = kvMapSavelocs->FindKey("startmarks");
+    if (!kvStartMarks)
+        return false;
+
+    CMomentumPlayer *pPlayer = CMomentumPlayer::GetLocalPlayer();
+    if (!pPlayer)
+        return false;
+
+    FOR_EACH_SUBKEY(kvStartMarks, kvStartMark)
+    {
+        int track = Q_atoi(kvStartMark->GetName());
+
+        SavedLocation_t * pStartmark = new SavedLocation_t;
+        pStartmark->Load(kvStartMark);
+
+        if (!pPlayer->SetStartMark(track, pStartmark))
+            return false;
```

Comment:
If it fails to set the start mark you're going to want to delete the created mark so it doesn't leak memory.

Reply:
This change is now implemented. A bit worrying that I missed that but I'm glad you caught it.